client/state/comments/reducer.js

```diff
@@ -126,7 +127,24 @@ export function totalCommentsCount( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Keeps track of if a comment list request is in flight
+ * @param {object} state  - global redux state
+ * @param {object} action - redux action
+ * @returns {boolean} new redux state
+ */
+export function isCommentListLoading( state = false, action ) {
+	switch ( action.type ) {
+		case COMMENTS_LIST_REQUEST:
+			return true;
+		case COMMENTS_RECEIVE: //we may need a more specific success/error action here
+			return false;
+	}
```

Comment:
yeah @gwwar I'm not feeling particularly strong about this.

the fact that we assume requests never fail is an entirely different problem that should be fixed and is a real anti-pattern. if we weren't reusing old code in the data layer comments stuff this wouldn't be a problem.

I just like rethinking about the problem and hopping that some of this repetition will positively influence the way we build new things and enhance old ones.